nixd/lib/Controller/Completion.cpp

```diff
@@ -298,7 +338,23 @@ void Controller::onCompletion(const CompletionParams &Params,
               completeAttrName(Scope, Prefix, Options,
                                ClientCaps.CompletionSnippets, List.items);
             }
-          } else {
+            FinishedCompletion = true;
+          }
+
+          // if we are in a literal path, use PathCompletionProvider only
+          if (!FinishedCompletion) {
```

Comment:
this boolean check looks ungly, what are we doing here?

What will happen if there is no this condition check?